add-extension.js

```diff
@@ -20,131 +20,150 @@ const { DH_UNABLE_TO_CHECK_GENERATOR } = require('constants');
 const readFile = util.promisify(fs.readFile);
 const writeFile = util.promisify(fs.writeFile);
 
-(async () => {
-  /**
-   * @type {{
-   *    extensions: {
-   *        id: string,
-   *        repository: string,
-   *        version?: string,
-   *        checkout?: string,
-   *        location?: string,
-   *        prepublish?: string,
-   *        extensionFile?: string,
-   *        download?: string
-   *    }[]
-   * }}
-   */
-  const { extensions } = JSON.parse(await readFile('./extensions.json', 'utf-8'));
-  const registry = new ovsx.Registry();
+const registry = new ovsx.Registry();
 
-  const argv = minimist(process.argv.slice(2));
-  if (argv._.length !== (!!argv.download ? 0 : 1)) {
-    console.log(`Usage: node add-extension REPOSITORY [OPTIONS]
-OPTIONS:
-    --checkout=CHECKOUT
-    --location=LOCATION
-    --prepublish=PREPUBLISH
-    --extensionFile=EXTENSION_FILE
-
-Alternative usage: node add-extension --download=VSIX_URL`);
-    process.exitCode = 1;
-    process.exit();
-  }
-
-  // Handle 'node add-extension --download=VSIX_URL':
-  if (argv.download) {
+// don't run this part if this module is not exported 
+// @ts-ignore
+if (!module.parent) {
```

Comment:
What is the exact meaning of this? Could you explain a bit more in the comment?

Reply:
@spoenemann it allows us to run the target code of `add-extension.js` if we don't want to use some function from this module, there is [an example](https://github.com/vymarkov/validate-vscode-extension-pack/blob/main/src/cli.ts#L7), I think, it better to extract the functions into file with name `utils.js` and then import them.

Reply:
Yes, better separate library code (reusable modules) from executable code. Could you move the functions you'd like to reuse elsewhere into a new file in the `lib` subdirectory?

Reply:
@spoenemann @jankeromnes I've just extracted the functions into lib directory